libparsec/crates/client/src/workspace/transactions/file_operations.rs

```diff
@@ -396,6 +396,22 @@ impl ReshapeBlockOperation<'_> {
         self.manifest_chunks
     }
 
+    pub fn is_pseudo_block(&self) -> bool {
+        self.manifest_chunks.len() == 1 && self.manifest_chunks[0].is_pseudo_block()
+    }
+
+    pub fn write_back(&self) -> bool {
+        !self.is_pseudo_block()
+    }
+
+    pub fn cleanup_ids(&self) -> HashSet<ChunkID> {
```

Comment:
The input data is a Vec, it is used by the caller as a Vec (i.e. for doing a loop)... would you consider using a Vec here :point_right: :point_left: (˶˃ᆺ˂˶)

Reply:
And I already updated the caller:
```rust
            for to_remove_chunk_id in reshape.cleanup_ids() {
                ...
```
I don't see the issue here.

Reply:
However, the hashset here is due to a legacy interface, I'll turn it into a vec.

Reply:
Nevermind, there might be duplicates so it's good to remove them.